src/monitor-location-workspace/monitor-location-checks.service.ts

```diff
@@ -0,0 +1,167 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Logger } from '@us-epa-camd/easey-common/logger';
+import { IMPORT_CHECK_ERROR } from '../utils/error.const';
+import { DailyTestSummaryImportDTO } from '../dto/daily-test-summary.dto';
+import { EmissionsImportDTO } from '../dto/emissions.dto';
+import { HourlyOperatingImportDTO } from '../dto/hourly-operating.dto';
+import { SorbentTrapImportDTO } from '../dto/sorbent-trap.dto';
+import { WeeklyTestSummaryImportDTO } from '../dto/weekly-test-summary.dto';
+import { MonitorLocation } from '../entities/workspace/monitor-location.entity';
+import { LocationIdentifiers } from '../interfaces/location-identifiers.interface';
+
+import { MonitorLocationWorkspaceRepository } from './monitor-location.repository';
+
+// the following types have componentId field (and possibly other fields later on) which is needed for addLocation()
+type ForLocationType = HourlyOperatingImportDTO  | DailyTestSummaryImportDTO | WeeklyTestSummaryImportDTO | SorbentTrapImportDTO
+
+@Injectable()
+export class MonitorLocationChecksService {
+  constructor(
+    private readonly logger: Logger,
+    @InjectRepository(MonitorLocationWorkspaceRepository)
+    private readonly repository: MonitorLocationWorkspaceRepository,
+  ) {}
+
+  processLocations(payload: EmissionsImportDTO): LocationIdentifiers[] {
+    const locations: LocationIdentifiers[] = [];
+
+    const addLocation = (i: ForLocationType) => {
+
+      const location = locations.find(
+        l => l?.unitId === i?.unitId || l?.stackPipeId === i?.stackPipeId,
+      );
+
+      if (location) {
+        if ("monitorHourlyValueData" in i)
+          (i as HourlyOperatingImportDTO).monitorHourlyValueData.forEach(d =>{
```

Comment:
Feel free to ignore this entire comment because it's a Monday.

The 'as' here is unnecessary, TypeScript is able to infer here. The screenshot is from the SonarLint plugin, which can be connected to our Sonar org for any custom rules.

![2022-08-29 09_32_16-easey-emissions-api – monitor-location-checks service ts Administrator](https://user-images.githubusercontent.com/3924176/187226012-5c4322c3-757a-476a-a4d1-65f4c05396b4.png)

Also on a note with i's, j's, d's, etc. I find these less readable. I obsessively name everything, but for something like a one-line map or filter, it's fine. For a function like this, I feel like 'i' may be better as 'forLocation' for some readability.

This will also reduce your own cognitive load when writing out logic like this.

Reply:
Yeah, this was also mistakenly left there. VSCode was being weird and showing linting issues there hence why I was going a little crazy and playing with 'as' there. Turned out i just needed to restart vscode but forgot to remove that bit